ground/src/main/java/com/google/android/ground/model/geometry/Point.kt

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.ground.model.geometry
+
+import org.locationtech.jts.geom.Coordinate
+import org.locationtech.jts.geom.GeometryFactory
+import org.locationtech.jts.geom.Point
+
+/** A single point given by x and y coordinates in a Cartesian plane. */
+data class Point(val x: Double, val y: Double) : Geometry {
```

Comment:
Why do we need these adapter classes? Can we just reference JTS Geometry directly in LOI?

Reply:
We could, but there are some benefits to wrapping:

1. For comparisons, we want `data class` (a.k.a AutoValue) behavior. e.g. two points are equal by value, not by reference. (at least this was our previous behavior and it seems correct).
2. If we decide JTS doesn't work down the line, we can swap it out for a different underlying library (though this is unlikely, from an OSS perspective it also gives forks the opportunity to do this).
3. The wrapper classes give us a convenient place to provide utilities JTS doesn't provide as built-ins (e.g. if a PolyLine is a Ring, convert it to a Polygon).
4. When it comes to converting coordinates to map impl representations (e.g. LatLng) these classes give us a convenient boundary for middle-man conversion to a general base-type representation (Coordinate -> Double), v.s. needing to implement this logic for every Map, we only implement the second half (Java type to Map types) (Double -> LatLng).

That said, we could just use them directly. Based on working with the code so far, this feels a bit cleaner. LOI and the rest of the application doesn't need to know anything about JTS.

Reply:
That all makes sense, though then it begs to question - if we have our own representations, why use JTS at all? GeoJSON (de)serialization is easy enough to implement if we're only defining a small subset of geometries. My understanding was that we'd use JTS as if it were a standard library like Java/Kotlin Collections to simplify the model.

One alternative could be to use Kotlin extension functions to simplify JTS usage. Let's discuss offline.